dokka-runners/dokka-gradle-plugin/src/testFixtures/kotlin/GradleTestKitUtils.kt

```diff
@@ -0,0 +1,349 @@
+          |
+        """.trimMargin()
+
+        build()
+    }
+}
+
+/**
+ * Builder for testing a Gradle project that uses Groovy script and creates default,
+ * `settings.gradle`, and `gradle.properties` files.
+ *
+ * @param[testProjectName] the name of the test, which should be distinct across the project
+ */
+fun gradleGroovyProjectTest(
+    testProjectName: String,
+    baseDir: Path = GradleProjectTest.funcTestTempDir,
+    build: GradleProjectTest.() -> Unit,
+): GradleProjectTest {
+    return gradleProjectTest(
+        testProjectName = testProjectName,
+        baseDir = baseDir,
+    ) {
+        gradleProperties += """
+            |org.jetbrains.dokka.experimental.gradlePlugin=dokkatoo!
+        """.trimMargin()
+
+        settingsGradle = """
+          |rootProject.name = "test"
+          |
+          |${settingsRepositories()}
+          |
+        """.trimMargin()
+
+        build()
+    }
+}
+
+private fun gradleProjectTest(
+    testProjectName: String,
+    baseDir: Path = GradleProjectTest.funcTestTempDir,
+    build: GradleProjectTest.() -> Unit,
+): GradleProjectTest {
+    return GradleProjectTest(baseDir = baseDir, testProjectName = testProjectName).apply {
+
+        gradleProperties = """
+      |kotlin.mpp.stability.nowarn=true
+      |org.gradle.cache=true
+      |
+    """.trimMargin()
+
+        build()
+    }
+}
+
+
+fun GradleProjectTest.projectFile(
+    @Language("TEXT")
+    filePath: String
+): PropertyDelegateProvider<Any?, ReadWriteProperty<Any?, String>> =
+    PropertyDelegateProvider { _, _ ->
+        TestProjectFileProvidedDelegate(this, filePath)
+    }
+
+
+/** Delegate for reading and writing a [GradleProjectTest] file. */
+private class TestProjectFileProvidedDelegate(
+    private val project: GradleProjectTest,
+    private val filePath: String,
+) : ReadWriteProperty<Any?, String> {
+    override fun getValue(thisRef: Any?, property: KProperty<*>): String =
+        project.projectDir.resolve(filePath).readText()
+
+    override fun setValue(thisRef: Any?, property: KProperty<*>, value: String) {
+        project.createFile(filePath, value)
+    }
+}
+
+/** Delegate for reading and writing a [GradleProjectTest] file. */
+class TestProjectFileDelegate(
+    private val filePath: String,
+) : ReadWriteProperty<ProjectDirectoryScope, String> {
+    override fun getValue(thisRef: ProjectDirectoryScope, property: KProperty<*>): String =
+        thisRef.projectDir.resolve(filePath).readText()
+
+    override fun setValue(thisRef: ProjectDirectoryScope, property: KProperty<*>, value: String) {
+        thisRef.createFile(filePath, value)
+    }
+}
+
+
+@DslMarker
+annotation class ProjectDirectoryDsl
+
+@ProjectDirectoryDsl
+interface ProjectDirectoryScope {
+    val projectDir: Path
+
+//    val testMavenRepoRelativePath: String
+//        get() = projectDir
+//            .relativize(GradleProjectTest.testMavenRepoDir)
+//            .invariantSeparatorsPathString
```

Comment:
Shouldn't it be used?

Reply:
See https://github.com/Kotlin/dokka/pull/3695#discussion_r1705325594